futures-core/src/task/context.rs

```diff
@@ -110,6 +110,19 @@ if_std! {
                 .expect("No default executor found")
                 .spawn(Box::new(f)).unwrap()
         }
+
+        /// Produce a context like the current one, but using the given executor
+        /// instead.
+        ///
+        /// This advanced method is primarily used when building "internal
+        /// schedulers" within a task.
+        pub fn with_executor<'b>(&'b mut self, executor: Option<&'b mut Executor>)
```

Comment:
IMO this should take the executor directly instead of an `Option`: the executor value should only ever be `None` when using the no-std version.

Reply:
Ah, okay! I think I was just a little confused by the fact that the `Context::executor` method is also flagged `if_std!` and returns an `Option`. Will change this!